survive-the-monsters.js

```diff
@@ -142,17 +196,24 @@ function spawnEnemy(x, y, type) {
         }
     }
     enemiesLargestId = Math.max(enemiesLargestId, enemyId);
-
-    enemiesState[enemyId] = 1;
+    if (enemiesLargestId == ENEMIES_COUNT_MAX) {
+        winGame();
+        return;
+    }
+    enemiesState[enemyId] = enemyHealth;
     enemiesType[enemyId] = type;
     enemiesX[enemyId] = x;
     enemiesY[enemyId] = y;
     actors.enemies[enemyId] = scenes.main.physics.add.image(SPRITE_SIZE, SPRITE_SIZE, `monster_${type}`);
+    actors.enemies[enemyId].setName(enemyId);
     actors.enemies[enemyId].setCircle(SPRITE_SIZE / 2, 0, 0);
-    scenes.main.physics.add.collider(actors.player, actors.enemies[enemyId], endGame);
+    scenes.main.physics.add.collider(actors.player, actors.enemies[enemyId], loseGame);
     actors.enemies[enemyId].setPosition(x, y);
 }
 function despawnEnemy(enemyId) {
```

Comment:
**praise**: Great use of Phaser's collider to handle bullet and enemy interactions. This keeps the code clean and leverages Phaser's built-in capabilities.